File: C#/MatrixMultiply/MatrixProduct.cs
using System;

class MatrixProduct
{
    static void Main(string[] args)
    {
        int op = 1;
        var instance = new Matrix();

        do
        {
            Console.WriteLine();
            Console.WriteLine("-------------------------");
            Console.WriteLine("1. Multiplication");
            Console.WriteLine("2. Line Multiplication");
            Console.WriteLine("3. Parallel Multiplication");
            Console.WriteLine("4. Parallel Line Multiplication");
            Console.WriteLine();
            Console.WriteLine("0. Exit");
            Console.WriteLine("-------------------------");
            Console.Write("Selection: ? ");

            string input = Console.ReadLine();

            if (int.TryParse(input, out op))
            {
                int numberLines;
                int numberColumns;

                if (op == 1 || op == 2 || op == 3 || op == 4)
                {
                    Console.Write("Dimensions: lins cols ? ");

                    string[] tokens = Console.ReadLine().Split();

                    if (tokens.Length == 2 && int.TryParse(tokens[0], out numberLines) && int.TryParse(tokens[1], out numberColumns))
                    {
                        switch (op)
                        {
                            case 1:
                                instance.Multiply(numberLines, numberColumns);
                                break;
                            case 2:
                                instance.MultiplyLine(numberLines, numberColumns);
                                break;
                            case 3:
                                instance.MultiplyParallel(numberLines, numberColumns);
                                break;
                            case 4:
                                instance.MultiplyLineParallel(numberLines, numberColumns);
                                break;
                        }
                    }
                }
            }
            else
            {
                Console.WriteLine("Error: non-numeric input is unsupported, try again.");
                op = 1;
            }
        } while (op != 0);

        Console.WriteLine();
    }
}

File: Java/src/cpar_proj1/MatrixProduct.java
package cpar_proj1;

import java.util.Scanner;

public class MatrixProduct
{
	private static Matrix instance = new Matrix();
	private static Scanner scanner = new Scanner(System.in);

	public static void main(String[] args)
	{
		int userChoice = 1;

		do
		{
			System.out.println("+---------------------------------------+");
			System.out.println("| 1. Multiplication                     |");
			System.out.println("| 2. Line Multiplication                |");
			System.out.println("| 3. Parallel Multiplication            |");
			System.out.println("| 4. Parallel Line Multiplication       |");
			System.out.println("+---------------------------------------+");
			System.out.println("| 0. Exit                               |");
			System.out.println("+---------------------------------------+");
			System.out.println();
			System.out.print("Selection: ? ");

			final String userString = scanner.nextLine();

			try
			{
				userChoice = Integer.parseInt(userString);

				if (userChoice > 0 && userChoice < 5)
				{
					System.out.print("Dimensions: lins cols ? ");

					final String[] tokens = scanner.nextLine().split(" ");

					try
					{
						int numberLines = Integer.parseInt(tokens[0]);
						int numberColumns = Integer.parseInt(tokens[1]);

						if (tokens.length == 2)
						{
							switch (userChoice)
							{
							case 1:
								instance.Multiply(numberLines, numberColumns);
								break;
							case 2:
								instance.MultiplyLine(numberLines, numberColumns);
								break;
							case 3:
								instance.MultiplyParallel(numberLines, numberColumns);
								break;
							case 4:
								instance.MultiplyLineParallel(numberLines, numberColumns);
								break;
							}
						}
					}
					catch (final NumberFormatException ex)
					{
						System.out.println("Error: non-numeric input is unsupported, try again.");
						userChoice = 1;
					}
				}
				else
				{
					System.out.println("Error: non-numeric input is unsupported, try again.");
					userChoice = 1;
				}
			}
			catch (final NumberFormatException ex)
			{
				System.out.println("Error: non-numeric input is unsupported, try again.");
				userChoice = 1;
			}
		} while (userChoice != 0);
	}
}